### PR TITLE
[FSTORE-1281] Add method to stop execution + type hinting login+get_feature_store

### PR DESCRIPTION
**Pull request by vatj**
…store + stop execution method

### CODE REVIEW

**Review by robzor92 — CHANGES_REQUESTED** [review]
Looks good, but need to fix linter.
Also if we could add another test here with a workflow where the execution gets stopped would be great: https://github.com/logicalclocks/loadtest/blob/main/tests/workflows/hopsworks/test_job_execution.py

File: python/hopsworks/project.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ def created(self):
         """Timestamp when the project was created"""
         return self._created
 
-    def get_feature_store(self, name: str = None):
+    def get_feature_store(self, name: str = None) -> feature_store.FeatureStore:
```
Comment (robzor92) [inline python/hopsworks/project.py:106]: Does it not need also the module here? `hsfs.feature_store.FeatureStore`
https://docs.hopsworks.ai/feature-store-api/latest/generated/api/feature_store_api/

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: python/hopsworks/project.py (hunk 1 above)
Comment (vatj) [reply python/hopsworks/project.py:106]: It is imported above from hsfs import feature_store

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,23 +12,24 @@
     pip install -e ".[dev]"
     ```
 
-- Install [pre-commit](https://pre-commit.com/) and then activate its hooks. pre-commit is a framework for managing and maintaining multi-language pre-commit hooks. The library uses pre-commit to ensure code-style and code formatting through [black](https://github.com/psf/black) and [flake8](https://gitlab.com/pycqa/flake8). Run the following commands from the `python` directory:
+- Install [pre-commit](https://pre-commit.com/) and then activate its hooks. pre-commit is a framework for managing and maintaining multi-language pre-commit hooks. The Feature Store uses pre-commit to ensure code-style and code formatting through [ruff](https://docs.astral.sh/ruff/). Run the following commands from the `python` directory:
```
Comment (robzor92) [inline CONTRIBUTING.md:15]:
```suggestion
- Install [pre-commit](https://pre-commit.com/) and then activate its hooks. pre-commit is a framework for managing and maintaining multi-language pre-commit hooks. The library uses pre-commit to ensure code-style and code formatting through [ruff](https://docs.astral.sh/ruff/). Run the following commands from the `python` directory:
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -12,23 +12,24 @@
     pip install -e ".[dev]"
     ```
 
-- Install [pre-commit](https://pre-commit.com/) and then activate its hooks. pre-commit is a framework for managing and maintaining multi-language pre-commit hooks. The library uses pre-commit to ensure code-style and code formatting through [black](https://github.com/psf/black) and [flake8](https://gitlab.com/pycqa/flake8). Run the following commands from the `python` directory:
+- Install [pre-commit](https://pre-commit.com/) and then activate its hooks. pre-commit is a framework for managing and maintaining multi-language pre-commit hooks. The Feature Store uses pre-commit to ensure code-style and code formatting through [ruff](https://docs.astral.sh/ruff/). Run the following commands from the `python` directory:
```
Comment (robzor92) [inline docs/CONTRIBUTING.md:15]:
```suggestion
- Install [pre-commit](https://pre-commit.com/) and then activate its hooks. pre-commit is a framework for managing and maintaining multi-language pre-commit hooks. The library uses pre-commit to ensure code-style and code formatting through [ruff](https://docs.astral.sh/ruff/). Run the following commands from the `python` directory:
```

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (robzor92) [reply CONTRIBUTING.md:15]: tiny suggestion since this is not the hsfs library

Comment (vatj) [reply CONTRIBUTING.md:15]: Ah yes perfect thanks for catching this!

**Review by robzor92 — APPROVED** [review]

**Pull request merged**